mozci/data/sources/artifact/__init__.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+import json
+from collections import defaultdict
+from typing import Any, Dict
+
+import requests
+
+from mozci.data.base import DataSource
+from mozci.util.taskcluster import get_artifact, list_artifacts
+
+
+class ErrorSummarySource(DataSource):
```

Comment:
We could potentially merge this into the Taskcluster datasource, and provide a way to prioritize contracts and data sources idependently, butI think it's not worth the additional complexitiy.

Reply:
Wouldn't be a bad feature.. but yeah definitely out of scope for this PR